packages/flutter/test/widgets/widget_inspector_test.dart

```diff
@@ -2057,46 +2075,239 @@ class _TestWidgetInspectorService extends TestWidgetInspectorService {
             expect(childrenFromOtherApi.length, equals(children.length));
           }
 
-          testWidgets('ext.flutter.inspector.getRootWidgetSummaryTree',
+          bool allChildrenSatisfyCondition(Map<String, Object?> treeRoot,
+              {required bool Function(Map<String, Object?> child) condition}) {
+            final List<Object?> children = childrenFromJsonResponse(treeRoot);
+            if (children.isEmpty) {
+              return true;
+            }
```

Comment:
we don't need this extra check since the for loop will be a no-op if the list is empty, and we will progress to the next `return true` statement after the loop

Reply:
ahh good point!